kcidb/db/postgresql/v04_09.py

```diff
@@ -0,0 +1,134 @@
+    io = io.schema.V4_5
+
+    # A map of table names and Table constructor arguments
+    # For use by descendants
+    # Add environment_compatible, and number_* fields
+    TABLES_ARGS = merge_dicts(
+        PreviousSchema.TABLES_ARGS,
+        checkouts=merge_dicts(
+            PreviousSchema.TABLES_ARGS["checkouts"],
+            columns=merge_dicts(
+                PreviousSchema.TABLES_ARGS["checkouts"]["columns"],
+                {
+                    "git_commit_tags": TextArrayColumn(),
+                    "git_commit_message": TextColumn(),
+                    "git_repository_branch_tip": BoolColumn(),
+                }
+            )
+        )
+    )
+
+    # A map of table names and schemas
+    TABLES = {
+        name: Table(**args) for name, args in TABLES_ARGS.items()
+    }
+
+    # A map of index names and schemas
+    INDEXES = merge_dicts(PreviousSchema.INDEXES, dict(
+        checkouts_git_commit_tags=Index(
+            "checkouts", ["git_commit_tags"]
+        ),
+        checkouts_git_repository_branch_tip=Index(
+            "checkouts", ["git_repository_url"]
+        ),
+    ))
+
+    # Queries and their columns for each type of raw object-oriented data.
+    # Both should have columns in the same order.
+    OO_QUERIES = merge_dicts(
+        PreviousSchema.OO_QUERIES,
+        checkout=dict(
+            statement="SELECT\n"
+                      "    id,\n"
+                      "    git_commit_hash,\n"
+                      "    git_commit_tags,\n"
+                      "    git_commit_message,\n"
+                      "    patchset_hash,\n"
+                      "    origin,\n"
+                      "    git_repository_url,\n"
+                      "    git_repository_branch,\n"
+                      "    git_repository_branch_tip,\n"
+                      "    tree_name,\n"
+                      "    message_id,\n"
+                      "    start_time,\n"
+                      "    log_url,\n"
+                      "    log_excerpt,\n"
+                      "    comment,\n"
+                      "    valid,\n"
+                      "    misc\n"
+                      "FROM checkouts",
+            schema=Table(dict(
+                id=TextColumn(),
+                git_commit_hash=TextColumn(),
+                git_commit_tags=TextArrayColumn(),
+                git_commit_message=TextColumn(),
+                patchset_hash=TextColumn(),
+                origin=TextColumn(),
+                git_repository_url=TextColumn(),
+                git_repository_branch=TextColumn(),
+                git_repository_branch_tip=BoolColumn(),
+                tree_name=TextColumn(),
+                message_id=TextColumn(),
+                start_time=TimestampColumn(),
+                log_url=TextColumn(),
+                log_excerpt=TextColumn(),
+                comment=TextColumn(),
+                valid=BoolColumn(),
+                misc=JSONColumn(),
+            )),
```

Comment:
Could you avoid duplicating the query schema here? Instead inherit it the same way `v04_01.py` does. And we would also need to fix `v04_08.py` to do the same.

Reply:
Updated it.